crowbar_framework/app/controllers/backups_controller.rb

```diff
@@ -0,0 +1,171 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class BackupsController < ApplicationController
+  skip_before_filter :enforce_installer
+  before_action :set_backup, only: [:destroy, :restore, :download]
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup 	GET 	Returns a json list of available backups
+  def index
+    @backups = Backup.all
+
+    respond_to do |format|
+      format.html
+      format.json { render json: @backups }
+    end
+  end
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup   POST   Trigger a backup
+  def create
+    @backup = Backup.new(backup_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        msg = I18n.t("backups.index.invalid_filename")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    respond_to do |format|
+      if @backup.restore
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json do
+          render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = @backup.errors
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    respond_to do |format|
+      if @backup.exist?
+        format.any do
+          send_file(
+            @backup.path,
+            filename: @backup.filename
+          )
+        end
+      else
+        msg = I18n.t("backups.index.missing_backup")
+        format.json { render json: { error: msg }, status: :not_found }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Upload
+  #
+  # Provides the restful api call for
+  # /utils/backup/upload   POST   Upload a backup
+  def upload
+    @backup = Backup.new(backup_upload_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json do
+          render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = @backup.errors
```

Comment:
Wrong... This should be some generic i18n message like on create. `errors` is a hash of error messages, so this should never work.